From d7b246e19d4e81f231b3aff6c3885c325be9a9d2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@redhat.com>
Date: Wed, 13 Dec 2017 13:38:41 +0100
Subject: [PATCH 10/41] i386: expose fw_cfg QEMU0002 in SSDT
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

RH-Author: Marc-André Lureau <marcandre.lureau@redhat.com>
Message-id: <20171213133912.26176-11-marcandre.lureau@redhat.com>
Patchwork-id: 78360
O-Subject: [RHEL-7.5 qemu-kvm PATCH v3 10/41] i386: expose fw_cfg QEMU0002 in SSDT
Bugzilla: 1411490
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

RHEL only: This is needed so kernel module can find the device and
load the driver. Upstream qemu uses different API to build ACPI
tables.

Signed-off-by: Marc-André Lureau <marcandre.lureau@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/ssdt-misc.dsl | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/hw/i386/ssdt-misc.dsl b/hw/i386/ssdt-misc.dsl
index a4484b8..d124c74 100644
--- a/hw/i386/ssdt-misc.dsl
+++ b/hw/i386/ssdt-misc.dsl
@@ -74,6 +74,16 @@ DefinitionBlock ("ssdt-misc.aml", "SSDT", 0x01, "BXPC", "BXSSDTSUSP", 0x1)
     External(\_SB.PCI0, DeviceObj)
     External(\_SB.PCI0.ISA, DeviceObj)
 
+    Scope(\_SB.PCI0) {
+        Device(FWCF) {
+            Name(_HID, "QEMU0002")
+            Name(_STA, 0x0B)
+            Name(_CRS, ResourceTemplate () {
+                IO(Decode16, 0x0510, 0x0510, 0x01, 0x0C, IO)
+            })
+        }
+    }
+
     Scope(\_SB.PCI0.ISA) {
         Device(PEVT) {
             Name(_HID, "QEMU0001")
-- 
1.8.3.1

